src/backend/marsha/core/tests/test_api_video.py

```diff
@@ -1630,6 +1658,28 @@ def test_api_video_update_detail_anonymous(self):
         video.refresh_from_db()
         self.assertEqual(video.title, "my title")
 
+    def test_api_video_update_detail_student(self):
+        """Student users should not be allowed to update a video through the API."""
+        video = factories.VideoFactory(title="my title")
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = ["student"]
+
+        data = {"title": "my new title"}
```

Comment:
I'm not sure a `put` with partial data would work for an instructor though?

Reply:
No, it doesn't see `test_api_video_put_by_playlist_instructor`, but it does with `test_api_video_put_by_organization_admin`

Reply:
What I meant is that a put should necessarily receive a full object to work. The best way to write this test would be to make it work with an admin and then change only the person who submits the data and see that it then fails with a 403?
Said differently: the test as it is would not work because the data submitted is not valid, even if the person had permissions.

Reply:
Sorry, I was not clear : the same payload is used in `test_api_video_put_by_organization_admin` :
```python
    def test_api_video_put_by_organization_admin(self):
        """Organization admins can update videos on the API."""
        user = factories.UserFactory()
        organization = factories.OrganizationFactory()
        factories.OrganizationAccessFactory(
            role=models.ADMINISTRATOR, organization=organization, user=user
        )
        playlist = factories.PlaylistFactory(organization=organization)
        video = factories.VideoFactory(playlist=playlist, title="existing title")

        jwt_token = AccessToken()
        jwt_token.payload["resource_id"] = str(user.id)
        jwt_token.payload["user"] = {
            "id": str(user.id),
            "username": user.username,
        }

        response = self.client.put(
            f"/api/videos/{video.id}/",
            json.dumps({"title": "updated title"}),
            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
            content_type="application/json",
        )

        self.assertEqual(response.status_code, 200)
        video.refresh_from_db()
        self.assertEqual(video.title, "updated title")
        ```

Reply:
:open_mouth: maybe because it's only required field :man_shrugging: